site/developer/developer_guide.md

```diff
@@ -0,0 +1,238 @@
+
+### Keys for Top Level and `folders` Elements
+
+These JSON keys are applicable for the top-level element (such as `JDBCSystemResource`), and each of its nested `folders` elements (such as `JdbcResource` and `JdbcResource/JDBCConnectionPoolParams`.
+
+#### `wlst_type`
+
+This value is the type name of the WLST MBean that corresponds to a model folder. The `${x:y}` notation described above is often used here to distinguish offline and online folder names.
+
+#### `child_folders_type`
+
+This value specifies how the tool will map the domain model element to one or more WLST MBeans. The values are:
+
+- `single` (default) - this element represents a single MBean, and the MBean name is known.
+
+- `single-unpredictable` - this element represents a single MBean, but the MBean name must be derived at runtime.
+
+- `multiple` - this element contains multiple named elements (such as `dataSource1`, `dataSource2`), and each represents a single MBean. 
+
+#### `folders`
+
+Nested WLST MBean types for the current MBean are listed here. Each has a domain model type name, followed by its own JSON keyed elements.
+ 
+#### `wlst_attributes_path`
+
+This key element specifies the name of the path expression used for navigating to the MBean attributes folder. The actual path expression is defined later in the `"wlst_paths": { }` element. 
+
+#### `wlst_paths`
+
+The dictionary key defines the various WLST path values used elsewhere in this folder's definition. Each entry maps a name to a full WLST MBean path.  In this example, `JDBCConnectionPoolParams` has a single path:
+
+`"WP001": "/JDBCSystemResource${:s}/%DATASOURCE%/${Jdbc:JDBC}Resource/%DATASOURCE%/JDBCConnectionPoolParams/${NO_NAME_0:%DATASOURCE%}"`
+
+The `%DATASOURCE%` text is token placeholder. It will be replaced with the name of the actual datasource by the tool.
+
+### Keys for `attributes` Elements
+
+Each child of an `attributes` element represents a single MBean attribute, and its key is the corresponding model name, such as `CapacityIncrement`. It contains at least one description element with the JSON keys below. There may be multiple description elements for cases where the configuration varies for different WebLogic Server version ranges, or varies between offline and online WLST.
+
+#### `version`
+
+This key element defines the applicable versions for a particular MBean attribute description. Maven versioning conventions are used to describe ranges and limits. For example:
+ 
+`"version": "[10,)"`
+  
+Specifies that an MBean attribute description is relevant for WebLogic Server version 10 and higher
```

Comment:
higher -> later. (period) (version – describe as a later version – not a higher version.)